src/rust/engine/process_execution/src/remote.rs

```diff
@@ -93,27 +98,31 @@ fn make_execute_request(
 }
 
 fn extract_execute_response(
-  operation: &bazel_protos::operations::Operation,
-) -> Result<Option<ExecuteProcessResult>, String> {
+  mut operation: bazel_protos::operations::Operation,
+) -> Result<ExecuteProcessResult, ExecutionError> {
   if !operation.get_done() {
-    return Ok(None);
+    return Err(ExecutionError::NotFinished(operation.take_name()));
```

Comment:
... keeping a single string alive like this without cloning or reference counting is pretty neat. So micro.

Reply:
Yeah, it thought it was cute :)